trace/trace.go

```diff
@@ -190,6 +255,12 @@ type Span struct {
 	trace        *Trace
 	eventLock    sync.Mutex
 	sendLock     sync.RWMutex
+	spanContext  *SpanContext
```

Comment:
I'm leery of having two copies of the span's metadata here - what do we do if the `spanContext` trace ID and the `Span`'s trace ID conflict?

Reply:
I think we need to tease out the responsibility of `Trace` and `Span` a little bit more here. Again, probably better with something in a doc that describes the relationship between each of these in the context of a Honeycomb event. I'll work on that.